### DISCUSSION TITLE
Very frequent websocket disconnection (round icon on top right of control panel turns red and says disconnected)

### DISCUSSION THREAD

**Discussion by alexruffell** [zwave-js/zwave-js-ui#2966] [Q&A]
I very often see Z-Wave JS UI’s websocket icon turn red indicating that the websocket is disconnected. This happens super frequently when I try to do anything in ZwaveJSUI from my phone (Samsung S20 Ultra) but I see it also happen on my desktop. It appears to interfere with whatever I am trying to do (inclusion, exclusion, etc.) forcing me to try again.

I am a Home Assistant user and I installed ZWaveJSUI as an addon. I can’t figure out what causes the websocket to keep disconnecting and I do n0to know how to troubleshoot the issue. I am not even sure why it appears to be way worse when I am on my phone.

Is it a software issue on the HA side? ZWaveJSUI side? A networking issue? Any pointer on what to focus on would be greatly appreciated.

In case it matters, my Home Assistant instance is installed in a Proxmox VM running on an i7 Lenovo TinyPC.

I posted on the HA forum and was suggested to ask here.

**Comment by robertsLando** [comment]
I know hass addon uses some proxy configuration to work, does this only happen from smartphones or also from PC?

Would be nice to have a chrome console log error (just press F12 to open dev tools) to see if at least we see what's the error causing the disconnection. The other alternative woould be to don't use the addon but install zwave-js-ui using snap or docker way, dunno if portainer could be an option for you. Maybe @kpine could tell you how

**Reply by kpine** [reply]
I think those are good suggestions. The goal would be to determine whether the add-on/HA components are related to the problem. Another option is to expose port 8091 (or whatever port is preferred) in the add-on configuration, and access ZUI through host:port, instead of the HA ingress.

**Comment by alexruffell** [comment]
@robertsLando I see this issue more frequently on my Android mobile phone to the point that I stopped using it to do anything related to ZWave JS UI on it... which means lugging around my laptop to do all includes/excludes/etc. Also, the UI which is awesome on my desktop, is really hard to use on the mobile (tons of scrolling needed). As for the proxy, I don't believe that setup is accessible to me. I tried using the console log and looking for errors when the websocket icon turns red but saw none. In the snapshot below I reloaded the page and before the devices appeared the icon turned red 3 or 4 times. 

<img width="734" alt="image" src="https://user-images.githubusercontent.com/4006944/220389944-8166ecbb-16f8-4b15-95d7-a9f135db1dad.png">

In case it matters, the warning was about some chrome extension:
<img width="709" alt="image" src="https://user-images.githubusercontent.com/4006944/220390705-94cec6f5-2a0f-45a3-b11d-1475f2289bf2.png">


@kpine If I could expose the add on without making significant changes to my setup, I would love to try. The only settings I see in the add on configuration are:

When I first opened it up, the network port was set to 3000 which I am guessing was incorrect as the default is for it to be disabled and even after changing it to 8091 and re-starting the add-on, zwave still works but I still can't load http://homeassistant:8091 or IP:8091 ... so I am not even sure what the setting is for.

<img width="841" alt="image" src="https://user-images.githubusercontent.com/4006944/220389142-06236dfb-8cf5-44e6-b1ca-1ac6f53771bc.png">

@robertsLando @kpine  It appears that portainer is highly discouraged by HA and was removed from HACS. I found a workaround to install it from another repo but need to tread carefully so I don't blow my system up so it may take me a little to familiarize with it :)

**Reply by kpine** [reply]
Sorry, looks like you can't expose the UI port, so that is not an option.

You don't need to expose port 3000, and you shouldn't do it, that is for the web-socket server.

**Comment by alexruffell** [comment]
Ok, I disabled it although I am confused on the difference between this setting and this other one:

<img width="751" alt="image" src="https://user-images.githubusercontent.com/4006944/220410568-274292f6-e7bf-430a-a940-88f33e3d74d2.png">

When I first enabled the one in the addon configuration I thought it needed to match the one in the settings tab. Could having both enabled on the same port cause issues?

**Reply by kpine** [reply]
The setting in ZUI is to configure the port ZUI listens on.

The setting in the add-on is to expose that port for access outside of the host. It's a general setting all add-ons have. There's no reason to expose the port for use within HA, unless you are running other instances of HA outside and want to connect the Z-wave integration to that instance.
